src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -52,16 +53,26 @@ public CreatedWorkspace createWorkspace(
 
   public WorkspaceDescription getWorkspace(String id, AuthenticatedUserRequest userReq) {
 
-    if (!samService.isAuthorized(
-        userReq.getRequiredToken(),
-        SamUtils.SAM_WORKSPACE_RESOURCE,
-        id,
-        SamUtils.SAM_WORKSPACE_READ_ACTION)) {
-      throw new SamUnauthorizedException(
-          "User " + userReq.getEmail() + " is not allowed to read workspace " + id);
-    }
-
+    samService.workspaceAuthz(userReq, id, SamUtils.SAM_WORKSPACE_READ_ACTION);
     WorkspaceDescription result = workspaceDao.getWorkspace(id);
     return result;
   }
+
+  public void deleteWorkspace(String id, String userToken) {
+
+    AuthenticatedUserRequest userReq = new AuthenticatedUserRequest().token(Optional.of(userToken));
+    samService.workspaceAuthz(userReq, id, SamUtils.SAM_WORKSPACE_WRITE_ACTION);
```

Comment:
There's a `delete` action in Sam for workspaces- we should probably use that here instead.